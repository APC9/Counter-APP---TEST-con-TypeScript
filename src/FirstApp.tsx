
interface Props{
  title: string;
  subTitle?: string;
  name?: string;
}

export const FirstApp = ({ title, subTitle='Sin subtitulo', name='Sin nombre' }:Props) => {
  
  return (
    <>
      <h1 data-testid="test-title"> { title } </h1>
      {/* <code>{ JSON.stringify( newMessage ) }</code> */}
      {/* <p>{ subTitle }</p> */}
      <p>{ subTitle }</p>
      <span>{ name }</span>
    </>
  )
}


