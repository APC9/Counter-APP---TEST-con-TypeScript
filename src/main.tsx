import React from 'react'
import ReactDOM from 'react-dom/client'
import { CounterApp } from './CounterApp'
import { FirstApp } from './FirstApp'

import './styles.css'


{/* <FirstApp title={'Hola, Soy Goku'}/> */}
ReactDOM.createRoot(document.getElementById('root') as HTMLElement).render(
  //<React.StrictMode>
    <CounterApp value={100}/>
  //</React.StrictMode>,
)
