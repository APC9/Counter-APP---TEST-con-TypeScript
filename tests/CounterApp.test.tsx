import React from 'react'; 
import { fireEvent, render, screen } from '@testing-library/react'
import {CounterApp} from '../src/CounterApp';

describe(' test <CounterApp />', ()=>{

  const value= 10;

  test('debe hacer Match con el Snapshot', () => {  
    const {container}=render(<CounterApp value={value}/>);
    expect(container).toMatchSnapshot();
  })

  test('Debe de mostrar el Valor inicial de 10', () => {
    render(<CounterApp value={value}/>);
    expect(screen.getByText(value) ).toBeTruthy();
    expect(screen.getByRole('heading', {level:2} ).innerHTML ).toContain(value.toString())
  })

  test('debe incrementar con el boton +1', () => { 
    render(<CounterApp value={value}/>);
    fireEvent.click(screen.getByText('+1'))

    expect(screen.getByText('11') ).toBeTruthy();
    expect(screen.getByRole('heading', {level:2} ).innerHTML ).toContain('11')
  })
  
  test('debe decrementar con el boton -1', () => { 
    render(<CounterApp value={value}/>);
    fireEvent.click(screen.getByText('-1'))
    
    expect(screen.getByText('9') ).toBeTruthy();
    expect(screen.getByRole('heading', {level:2} ).innerHTML ).toContain('9')
  })
  
  test('debe restaurar el valor inicial con el boton Reset', () => { 
    render(<CounterApp value={value}/>);
    fireEvent.click(screen.getByText('-1'))
    fireEvent.click(screen.getByText('-1'))
    fireEvent.click(screen.getByText('-1'))
    
    screen.debug()
    const btnReset = screen.getByRole('button', {name:'btn-reset'}) // <button aria-label="btn-reset">
    fireEvent.click(btnReset)
    
    expect(screen.getByText('10') ).toBeTruthy();
    expect(screen.getByRole('heading', {level:2} ).innerHTML ).toContain('10')
  })

})