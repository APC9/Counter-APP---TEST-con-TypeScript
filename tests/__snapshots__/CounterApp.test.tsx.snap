// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` test <CounterApp /> debe hacer Match con el Snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
     
    10
     
  </h2>
  <button>
     +1 
  </button>
  <button>
     -1 
  </button>
  <button
    aria-label="btn-reset"
  >
     Reset 
  </button>
</div>
`;
